pkg/kratos/service.go

```diff
@@ -0,0 +1,171 @@
+package kratos
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+	misc "github.com/canonical/identity-platform-login-ui/internal/misc/http"
+	"github.com/canonical/identity-platform-login-ui/internal/monitoring"
+	hClient "github.com/ory/hydra-client-go/v2"
+	kClient "github.com/ory/kratos-client-go"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	kratos KratosClientInterface
+	hydra  HydraClientInterface
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+// We override the type from the kratos sdk, as it does not
+// get marshalled correctly into json
+type ErrorBrowserLocationChangeRequired struct {
+	Error *kClient.GenericError `json:"error,omitempty"`
+	// Points to where to redirect the user to next.
+	RedirectBrowserTo *string `json:"redirect_browser_to,omitempty"`
+}
+
+func (s *Service) CheckSession(ctx context.Context, cookies []*http.Cookie) (*kClient.Session, http.Header, error) {
+	_, span := s.tracer.Start(ctx, "kratos.FrontendApi.ToSession")
+	defer span.End()
+
+	session, resp, err := s.kratos.FrontendApi().ToSession(
+		ctx,
+	).Cookie(
+		misc.CookiesToString(cookies),
+	).Execute()
+
+	if err != nil {
+		// TODO @shipperizer we shouldn't be logging this
+		s.logger.Debugf("full HTTP response: %v", resp)
+
+		return nil, nil, err
+	}
+	return session, resp.Header, nil
+}
+
+func (s *Service) AcceptLoginRequest(ctx context.Context, identityID string, lc string) (*hClient.OAuth2RedirectTo, http.Header, error) {
+	_, span := s.tracer.Start(ctx, "hydra.OAuth2Api.AcceptOAuth2LoginRequest")
+	defer span.End()
+
+	accept := hClient.NewAcceptOAuth2LoginRequest(identityID)
+	redirect_to, resp, err := s.hydra.OAuth2Api().AcceptOAuth2LoginRequest(ctx).
+		LoginChallenge(lc).
+		AcceptOAuth2LoginRequest(*accept).
+		Execute()
+
+	if err != nil {
+		// TODO @shipperizer we shouldn't be logging this
+		s.logger.Debugf("full HTTP response: %v", resp)
+		return nil, nil, err
+	}
+
+	return redirect_to, resp.Header, nil
+}
+
+func (s *Service) CreateBrowserLoginFlow(
+	ctx context.Context, aal string, return_to string, login_challenge string, refresh bool, cookies []*http.Cookie,
```

Comment:
```suggestion
	ctx context.Context, aal, return_to, login_challenge string, refresh bool, cookies []*http.Cookie,
```

sugar syntax or shortcut (still kinda normal practice afaik), if subsequent args are the same type, u can specify only on the last one the type